android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -112,8 +111,10 @@ public void initKMKeyboard(final Context context) {
 
     setWebChromeClient(new WebChromeClient() {
       public boolean onConsoleMessage(ConsoleMessage cm) {
-        if (KMManager.isDebugMode())
-          Log.d("Keyman JS Log: Line " + cm.lineNumber(), cm.message());
+        if (KMManager.isDebugMode()) {
+          Log.d("Keyman JS Log: Line " + cm.lineNumber(), cm.sourceId() + ":" + cm.message());
+          Log.d("Keyman JS Log: Line " + cm.lineNumber(), cm.sourceId());
```

Comment:
Why the second line? Doesn't that just double the log messages?

Reply:
My mistake; I guess I never quite got that bit removed.